### PR TITLE
Update risk parity boxes factory

### PR DESCRIPTION
**Pull request by eirik-thorp**
Updates to RiskParityBoxesFactory:

- Change BloombergDataProvider to any class that implements AbstractDataProvider.
- Add functionality to define custom tickers.
- Replace inactive tickers.
- Replace NaN values before converting to simple returns.
- Update tests to reflect the changes above.

### CODE REVIEW

File: qf_lib/common/risk_parity_boxes/risk_parity_boxes.py [hunk 1]
```diff
@@ -89,15 +89,17 @@ class RiskParityBoxesFactory:
 
     Parameters
     ----------
-    bbg_data_provider: BloombergDataProvider
-        reference to bloomberg data provider
+    data_provider: AbstractPriceDataProvider
+        reference to a class that implements AbstractPriceDataProvider
```
Comment (myrmarachne) [inline qf_lib/common/risk_parity_boxes/risk_parity_boxes.py:93]: reference to the class providing the pricing data

File: qf_lib/common/risk_parity_boxes/risk_parity_boxes.py [hunk 2]
```diff
@@ -130,7 +132,7 @@ def _create_tickers_dict():
             ChangeDirection.RISING: {
                 ChangeDirection.RISING: [
                     BloombergTicker("SPGSCITR Index"),  # Commodities (S&P GSCI Total Return CME)
-                    BloombergTicker("MSBICBGU Index"),  # EM Debt (Morningstar Emerging Markets Corporate Bond Index TR)
+                    BloombergTicker("MSBICBGU Index"),  # EM Debt
```
Comment (myrmarachne) [inline qf_lib/common/risk_parity_boxes/risk_parity_boxes.py:135]: Morningstar Emerging Markets Corporate Bond GR USD

**Review by myrmarachne — CHANGES_REQUESTED** [review]

File: qf_lib/common/risk_parity_boxes/risk_parity_boxes.py (hunk 1 above)
Comment (eirik-thorp) [reply qf_lib/common/risk_parity_boxes/risk_parity_boxes.py]: Fixed

File: qf_lib/common/risk_parity_boxes/risk_parity_boxes.py (hunk 2 above)
Comment (eirik-thorp) [reply qf_lib/common/risk_parity_boxes/risk_parity_boxes.py]: Fixed

**Review by myrmarachne — APPROVED** [review]

**Pull request merged**